packages/rolldown/src/plugin/with-filter.ts

```diff
@@ -0,0 +1,42 @@
+import { HookFilterExtension, Plugin } from '..';
+
+type OverrideFilterObject = {
+  transform?: HookFilterExtension<'transform'>['filter'];
+  resolveId?: HookFilterExtension<'resolveId'>['filter'];
+  load?: HookFilterExtension<'load'>['filter'];
+};
+// TODO: maybe we should also change the name as well ?
+// my proposal is `${import.meta.filename}@${name}`
```

Comment:
I think it's better to keep the name. Because there's some plugins that checks the plugin name.
For example: https://github.com/vitejs/vite-plugin-react/blob/87f7fdd63d36e0317b4fa013bae3e15291153f28/packages/plugin-react-swc/src/index.ts#L120-L131

Reply:
ok